quinn-proto/src/connection/mod.rs

```diff
@@ -1878,15 +1881,18 @@ impl Connection {
         now: Instant,
         remote: SocketAddr,
         ecn: Option<EcnCodepoint>,
-        mut packet: Packet,
+        packet: Option<Packet>,
+        stateless_reset: bool,
```

Comment:
It looks like the `Option` wrapper around `Packet` and the `stateless_reset` flag are actually redundant state (and it looks like the `stateless_reset` flag is unused).

Reply:
For most stateless reset packets, `packet` will be `Some`. It is only `None` when the reset is shorter than the RFC recommends. The `stateless_reset` flag is read in the `Err(None)` case of `match decrypted`.

Reply:
Okay!